src/eureka/lib/util.py

```diff
@@ -637,3 +640,67 @@ def phot_arrays(data):
     data['aperr'].attrs['time_units'] = data.flux.attrs['time_units']
 
     return data
+
+
+def make_citations(meta, mods):
+    """Store relevant citation information in the current meta file.
+
+        Searches through imported libraries and current ECF parameters for 
+        terms that match BibTeX entries in citations.py. Every entry that 
+        matches gets added to a bibliography field in the meta file.
+    
+        Parameters
+        ----------
+        meta : eureka.lib.readECF.MetaClass
+            The current metadata object.
+
+        mods : array-like
+            Array of strings containing the currently installed modules.
+    """
+
+    # get modules for which we have citations, accessed in each Stage before 
+    # the call to make_citations
+    module_cites = np.intersect1d(mods, list(CITATIONS)).tolist()
+
+    # in S5, extract fitting methods/myfuncs to grab citations
+    other_cites = []
+    if hasattr(meta, 'fit_method'):
+        # need to remove citations to emcee/dynesty/batman/celerite/george if 
+        # they got picked up by the installed modules
+        for m in ['emcee', 'dynesty', 'george', 'celerite', 'batman']:
+            module_cites.remove(m)
+
+        if "emcee" in meta.fit_method:
+            other_cites.append("emcee")
+        if "dynesty" in meta.fit_method:
+            other_cites.append("dynesty")
+
+        # check if batman or GP is being used for transit/eclipse modeling
+        if hasattr(meta, "run_myfuncs"):
+            if "batman_tr" or "batman_ecl" in meta.run_myfuncs:
+                other_cites.append("batman")
+            if "GP" in meta.run_myfuncs:
+                if hasattr(meta, "GP_package"):
+                    other_cites.append(meta.GP_package) 
+
+    # I set the instrument in the relevant bits of S1/2, so I don't think this 
+    # should really be necessary. boilerplate for later
+    # if we already have the instrument, don't bother doing anything
+    if hasattr(meta, 'inst'):
```

Comment:
How does the `meta.inst` value even get used for citations?

Reply:
For example, if I run just Stage 6 then no MIRI ends up in my citations even though `meta.inst = 'miri'`